e2e/cypress/integration/system_console/team_and_channel_assign_roles_spec.js

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('System Console', () => {
+    it('MM-20059 - System Admin can map roles to groups from Team Configuration screen', () => {
+        cy.apiLogin('sysadmin');
+
+        // # Go to system admin page and to team configuration page of channel "eligendi"
+        cy.visit('/admin_console/user_management/teams');
+        cy.findByTestId('eligendiedit').click();
+
+        // # Wait until the groups retrieved and show up
+        cy.wait(500); //eslint-disable-line cypress/no-unnecessary-waiting
+
+        // # Remove all exisiting groups
+        cy.get('#groups-list--body').then((el) => {
+            if (el[0].childNodes[0].innerText !== 'No groups specified yet') {
+                for (let i = 0; i < el[0].childNodes.length; i++) {
+                    cy.get('#group-actions').click();
+                }
+
+                // # Save the setting
+                cy.get('#saveSetting').click();
+            }
+        });
```

Comment:
Any reason why we want to remove the existing groups? If the Groups are already available, then we can skip the step to add a group to this Team/Channel. If its not available, then we can add the first group in the group list as mentioned below.

Reply:
Already discussed but conclusion was that it is easier to remove the existing groups and add fresh cause then I don't need to add extra confusing code to see what the current settings are and save them and ensure it changed. If I remove and add from scratch, it ensures it's going from member to admin and back to admin. It simplifies the testing.